src/consumers/promise.js

```diff
@@ -0,0 +1,23 @@
+/* global setImmediate */
+const { pullBatch } = require("./common");
+
+
+// eslint-disable-next-line max-params
+function asyncBatch (gen, batchSize, arrayBuffer, resolve) {
+  const isLast = pullBatch(gen, batchSize, arrayBuffer);
+  if (isLast) {
+    resolve(arrayBuffer.join(""));
+  } else {
+    setImmediate(asyncBatch, gen, batchSize, arrayBuffer, resolve);
+  }
+}
+
+function toPromise (gen, batchSize) {
+  const arrayBuffer = [];
+  return new Promise(resolve =>
```

Comment:
Hmm... good question.  Since this is a simple promise, the performance difference is going to be negligible.  `bluebird` really shines when composing lots of promises, relying on them for everything async (like in [Interlock](http://www.interlockjs.com)).  So... probably fine?  I could make the change to the benchmark, see if it makes a difference.